src/Psalm/Internal/Provider/ReturnTypeProvider/StrReplaceReturnTypeProvider.php

```diff
@@ -1,62 +0,0 @@
-<?php
-namespace Psalm\Internal\Provider\ReturnTypeProvider;
-
-use function in_array;
-use PhpParser;
-use Psalm\CodeLocation;
-use Psalm\Context;
-use Psalm\StatementsSource;
-use Psalm\Type;
-
-class StrReplaceReturnTypeProvider implements \Psalm\Plugin\Hook\FunctionReturnTypeProviderInterface
-{
-    public static function getFunctionIds() : array
-    {
-        return [
-            'str_replace',
-            'str_ireplace',
-            'substr_replace',
-            'preg_replace',
-            'preg_replace_callback',
```

Comment:
Those are all defined in `stubs/CoreGenericFunctions.phpstub` now and have proper return types. I believe this file can be removed.

Reply:
See https://github.com/vimeo/psalm/pull/3503#issuecomment-643867424, there are still some tweaks in the Provider that can't be deleted

Reply:
Joy :-) - Guess this PR will be tougher then I anticipated :-)